File: RemoveElement.cpp
class Solution {
public:
    int removeElement(int A[], int n, int elem) {
        
        int temp;
        for (int i = 0; i < n; ++i) {
            if (A[i] == elem) {
                // swap with current end
                A[i] = A[n-1];
                --i;
                --n;
            }
        }
        return n;
    }
};

File: jiuzhang/remove-element/remove-element.py
"""
 * 本代码由九章算法编辑提供。没有版权欢迎转发。
 * - 九章算法致力于帮助更多中国人找到好的工作，教师团队均来自硅谷和国内的一线大公司在职工程师。
 * - 现有的面试培训课程包括：九章算法班，系统设计班，九章强化班，Java入门与基础算法班
 * - 更多详情请见官方网站：http://www.jiuzhang.com/
"""

class Solution:
    """
    @param A: A list of integers
    @param elem: An integer
    @return: The new length after remove
    """
    def removeElement(self, A, elem):
        # write your code here
        j = len(A)-1
        for i in range(len(A) - 1, -1, -1):
            if A[i] == elem:
                A[i], A[j] = A[j], A[i]
                j -= 1
        return j+1

